docs/data-model.md

```diff
@@ -0,0 +1,24 @@
+# Github for Jira Data Model
+
+![Data Model](./images/data-model.jpg)
+[draw.io file](./draw.io/data-model.drawio.xml)
+
+Github for Jira used to only integrate between Jira and github.com, but with the added support for Github Enterprise Server, we must be able to handle multiple Github apps to a single Jira instance.  This data model is to try to clarify the interactions between these entities and it's evolution as we continue to develop the integration.
```

Comment:
Should we throw a (coming soon) after GitHub Enterprise Server? Doubt many (any?) people will read this but if they do it would lead to confusion at this point while we're only supporting cloud.

Also: it's -> its (possessive pronoun)